packages/plugins/plugin-device-token/src/methods/___tests__/DeviceTokenPlugin.test.ts

```diff
@@ -0,0 +1,82 @@
+import { DeviceTokenPlugin } from '../../DeviceTokenPlugin';
+import { MockSegmentStore } from '../../../../../core/src/__tests__/__helpers__/mockSegmentStore';
+import { getMockLogger } from '../../../../../core/src/__tests__/__helpers__/mockLogger';
+import { SegmentClient } from '../../../../../core/src/analytics';
+import { Platform } from 'react-native';
+
+const mockRequestPermission = jest.fn().mockReturnValue(1);
+const mockGetAPNSToken = jest.fn();
+const mockGetDeviceToken = jest.fn();
+
+jest.mock('@react-native-firebase/messaging', () => () => ({
+  getAPNSToken: mockGetAPNSToken,
+  getToken: mockGetDeviceToken,
+  hasPermission: mockRequestPermission,
+}));
+
+describe('DeviceTokenPlugin', () => {
+  const store = new MockSegmentStore();
+  const clientArgs = {
+    logger: getMockLogger(),
+    config: {
+      writeKey: '123-456',
+      trackApplicationLifecycleEvents: true,
+    },
+    store,
+  };
+  let plugin: DeviceTokenPlugin = new DeviceTokenPlugin();
+
+  beforeEach(() => {
+    store.reset();
+    jest.clearAllMocks();
+    plugin = new DeviceTokenPlugin();
+  });
+
+  it('requests authorization when configure is called', async () => {
+    let configureSpy = jest.spyOn(plugin, 'configure');
+    let analytics = new SegmentClient(clientArgs);
+
+    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
+      OS: 'ios',
+      select: () => null,
+    }));
```

Comment:
Mocks should be at the top of the file because jest will actually execute them first (regardless of where you call them) (aka "hoist" them).

Also, I think we already have a global mock for `react-native`. Not sure if it has already a `Platform` object mocked but you can extend it